### PR TITLE
[bug 882558] Corrected error handling

### PR DESCRIPTION
**Pull request by sedge**
Ref: https://bugzilla.mozilla.org/show_bug.cgi?id=882558

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -82,14 +82,14 @@ module.exports = function ( app, rawUrl ) {
         uri: webmakerUrl + "isAdmin?id=" + id,
         json: true
       }, function ( error, response, body ) {
-        if ( response.statusCode == 401 ) {
-          return callback( "Authentication failed!" );
-        }
-
         if ( error || body.error ) {
```
Comment (jbuck) [inline index.js]: I wonder if it's safe to check body.error at this point

File: index.js [hunk 2]
```diff
@@ -82,14 +82,14 @@ module.exports = function ( app, rawUrl ) {
...
           return callback( error || body.error );
         }
 
+        if ( response.statusCode == 401 ) {
```
Comment (jbuck) [inline index.js:103]: You'll also need to change this code for the getUser method below

File: index.js (hunk 1 above)
Comment (sedge) [inline index.js]: I thought `body` always existed... but if it doesn't, then you're definitely right!

Comment (jbuck) [inline index.js]: I don't know, it may depend on the content-type?

Comment (sedge) [inline index.js]: I'll look into it.

Comment (sedge) [inline index.js]: Hmm... Its unclear when `request` will parse it. 

Could `|| body.error )` be changed to `|| !body )` or something similar? That might do it...

Comment (jbuck) [inline index.js]: Yes, you could check for the presence of body before looking for error, that'd be fine

File: index.js (hunk 2 above)
Comment (sedge) [inline index.js:103]: done

File: index.js [hunk 3]
```diff
@@ -60,12 +60,12 @@ module.exports = function ( app, rawUrl ) {
         uri: webmakerUrl + "user/" + id,
         json: true
       }, function ( error, response, body ) {
-        if ( response.statusCode == 401 ) {
-          return callback( "Authentication failed!" );
+        if ( error || ( body && body.error ) ) {
```
Comment (humphd) [inline index.js]: I think you should just pull these apart, and have an if for error, and another for body stuff.  Return early if there are errors, then keep going.  Don't combine so much like this, it's not helpful.

File: index.js [hunk 4]
```diff
@@ -82,12 +82,12 @@ module.exports = function ( app, rawUrl ) {
         uri: webmakerUrl + "isAdmin?id=" + id,
         json: true
       }, function ( error, response, body ) {
-        if ( response.statusCode == 401 ) {
-          return callback( "Authentication failed!" );
+        if ( error || ( body && body.error ) ) {
```
Comment (humphd) [inline index.js]: Same here.

**Pull request merged**